Draft release notes for 1.1.0 (#382)
apache/brooklyn-docs: guide/misc/release-notes.md

### New Features

#### Workflows

Apache Brooklyn now has a powerful workflow engine and syntax for defining entities, effectors, sensors, and policies.

> The syntax supports longhand, conditions, loops, error-handling, variables, a large set of built-in step types, and 

Reviewer: ```suggestion
It supports longhand and shorthand syntax, conditions, loops, error-handling, variables, a large set of built-in step types, and 
```

the ability to define custom step types.md).

```yaml
- type: some-entity
  brooklyn.initializers:
  - type: workflow-sensor
    brooklyn.config:
      sensor: count-how-often-other_sensor-is-published
      triggers:
        - other_sensor
      steps:
        - let integer x = ${entity.sensor.x} + 1 ?? 0
        - return ${x}
```

#### Workflow Enitity (workflow-entity)

Brooklyn now supports a `workflow-entity` where `start` / `stop` are defined by workflow.

#### Kubectl Task factory, Docker effector and Docker Sensor

> This is a practical and highly customizable way to externalize effectors and sensors to containers run on a Kubernetes cluster or docker container.

Reviewer: If Kubernetes is capitalised, Docker should too, or the opposite 
```suggestion
This is a practical and highly customizable way to externalize effectors and sensors to containers run on a Kubernetes cluster or Docker container.
```

services:
  - type: 'org.apache.brooklyn.entity.stock.BasicStartable:1.1.0-SNAPSHOT'
    brooklyn.initializers:
      - type: org.apache.brooklyn.tasks.kubectl.ContainerEffector
        brooklyn.config:
          name: container-effector
          description: Very simple container effector
          shell.env:
            hello: world-amp
          image: perl
          imagePullPolicy: IfNotPresent
          args:
            - echo
            - hello
```

#### Logbook Viewer

Logbook exposes through a new rest API endpoint the server logs. It’s packaged with two different logstore implementation:
- Static logfiles (default)
- A ElasticSearch instance

Logs can be seen in the about page but also as part of each task execution, as each task has it’s own ID. As task can create new sub-task, is possible to see the logs of all the child-task when filtering by the creator.

#### Updated to Apache Karaf 4.3.6

Karaf updates:

This release is an important release on the Karaf 4.3.x series containing:
- upgrade to Pax Logging 2.0.14 with log4j 2.17.1 (fixing CVE-2021-44832)
- prepare JDK 18 support
- fix deployment issue by upgrading to Apache Felix FileInstall 3.7.4
- and much more!

> - The Release Notes are available here: https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12311140&version=12351123

Reviewer: ```suggestion
The Release Notes are available here: https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12311140&version=12351123
```

#### Groups Change Policy

New policy for adding policies, enrichers, and initializers to entities as the join dynamic groups.

#### Connection tag

> Inspired by the Terraform `connection` element and added a `connection` tag that encapsulates connection details. When declared on an entity, any SSH steps will use the details to establish a connection and execute

Reviewer: ```suggestion
Inspired by the Hashicorp Terraform `connection` element and added a `connection` tag that encapsulates connection details. When declared on an entity, any SSH steps will use the details to establish a connection and execute
```

#### Persistence import/export API

Introducing an API for persistence import/export feature.
This is intended for file based persistence stores and as a parameter, it takes the location of root of the persistence store to be imported.

Invoking the operation will merge the new data to the currently existing store. The process is as follows:
- new temporary management context is created with the persistence store to be imported
- memento of that persistence store is captured
- bundles from the persistence store are installed in the active management context - this deals with bundles/types in the catalog and locations
- contents of relevant directories (policies, enrichers, etc). are written to the active management context. These are used for the deployed applications
- rebind method adds the deployed applications to the active management context without having to reset the full management context/restart the server

#### New ChildrenBatchEffector

Adding a new effector to call a inner effector in all the children entities where the effector is inserted in batches of a parametrized size.

> #### add a Secret object which can be used to capture a secret

Reviewer: ```suggestion
#### Add a Secret object which can be used to capture a secret
```

This can be used wherever we need extra assurance that credentials are not accidentally logged or shown in ui

### Security Fixes

> #### mitigate CVE-2023-1370

Reviewer: ```suggestion
#### Mitigate CVE-2023-1370
```
